ambry-rest/src/main/java/com.github.ambry.rest/NettyResponseChannel.java

```diff
@@ -342,6 +342,7 @@ private boolean maybeWriteResponseMetadata(HttpResponse responseMetadata,
       writtenThisTime = true;
       long writeProcessingTime = System.currentTimeMillis() - writeProcessingStartTime;
       nettyMetrics.responseMetadataProcessingTimeInMs.update(writeProcessingTime);
+      request.getMetricsTracker().nioMetricsTracker.markFirstByteSent();
```

Comment:
need to wrap this with a `if (request != null)`

Reply:
done